staging/src/k8s.io/component-base/logs/logs.go

```diff
@@ -76,3 +82,82 @@ func GlogSetter(val string) (string, error) {
 	}
 	return fmt.Sprintf("successfully set klog.logging.verbosity to %s", val), nil
 }
+
+const logLoggingFormat = "logging-format"
+
+var logRegistry *LogFormatRegistry = NewLogFormatRegistry()
+
+type LogFormatRegistry struct {
+	registry map[string]*logr.Logger
+	mu       sync.Mutex
+}
+
+func NewLogFormatRegistry() *LogFormatRegistry {
+	return &LogFormatRegistry{
+		registry: make(map[string]*logr.Logger),
+		mu:       sync.Mutex{},
```

Comment:
Not sure if registry should be thread safe as options are not.
@mtaufen do you have preference?